In [ ]:
import pandas as pd
import numpy as np

🐄 Suponha que foi sugerido que buracos negros de massa $10^{-8}M_{\odot}$ 
toda a matéria escura no halo de nossa Galáxia. 

  a) Quão longe seria
você espera que o buraco negro mais próximo seja? 

  b) Com que frequência seria
você espera que tal buraco negro passe dentro de 1 UA do Sol? (A
estimativa de ordem de grandeza é suficiente.)

🐄 Suponha que foi sugerido que MACHOs de massa $10^{-3}M_{\odot}$ (cerca de
massa de Júpiter) compunha toda a matéria escura no halo de nossa galáxia.

  c) A que distância você espera que o MACHO mais próximo esteja? 

  d) Quão freqüentemente tal MACHO passaria a 1 UA do Sol? (Novamente,
uma estimativa de ordem de grandeza será suficiente.)

➡ Supondo que a composição da matéria escura esteja relacionada com buracos negros. Calculamos a densidade dentro de um range para o raio efetivo variando entre $d=8200± 200 pc$.

➡ Dito isso, buscamos inicialmente determinar a densidade de matéria $ρ_{m} = \frac{ΔM(R)}{ΔV(R)}$ onde  $M(R)= \frac{v^{2}R}{G}$ sendo $v=235km/s$ e $G=4.3\cdot10^{-6}\frac{Kpc}{M_{⊙}}\frac{km^{2}}{s^{2}}$; e para o volume temos $ V(R)= \frac{4πR^{3}}{3}$ para acharmos a distância do buraco negro mais próximo.

In [ ]:
#iniciamos convertendo km/s para AU/yr, daí:

v = (235e3*3.156e7*206265)/(3.086e16)
print(v,'AU/yr')

49.57177572909916 AU/yr


In [ ]:
G= (6.67e-11*1.99e30)/(1e6*3.09e19)
G
#colocar as unidades necessárias no print

4.295566343042071e-06

In [ ]:
R=8.4
#é necessário fazer tais mudanças pois a velocidade está em AU/yr e a constante gravitacional está em km²/s²
M8400= (np.power(v,2)*R*(1.49e8)**2)/(G*(3.156e7)**2)
M8400

107109096356.08136

In [ ]:
R=8
M8000= (np.power(v,2)*R*(1.49e8)**2)/(G*(3.156e7)**2)
M8000

102008663196.26797

In [ ]:
deltaM= M8400-M8000
deltaM

5100433159.813385

In [ ]:
pi=3.14159265359

V8400= (4*pi*(8.4**3))/3
V8400

2482.712709537877

In [ ]:
V8000=(4*pi*(8**3))/3
V8000

2144.6605848507734

In [ ]:
deltaV=V8400-V8000
deltaV

338.0521246871035

In [ ]:
rho_m= (deltaM)/(deltaV)
print(rho_m,'Msol/Kpc³')

15087712.1820615 Msol/Kpc³


Daí, encontramos a densidade da matéria sendo $\rho ≈ 1.5 × 10^{7} M⊙/Ḱpc^{3}$

Buscamos alterar essa densidade para parsec, de maneira que eventualmente transformaremos para unidade astronômica pois o enunciado segue essa linha de raciocínio. Por isso fazemos $\frac{M_{⊙}}{(10^{3}pc)^{3}}$

In [ ]:
rho_m1= rho_m/1e9
rho_m1

0.015087712182061501

Dito isso, sabemos que a densidade de buracos negros é $\frac{2ρ_{m}}{3}$, então temos:

In [ ]:
rho_bn=(2/3)*rho_m1
rho_bn

0.010058474788041

Daí, a densidade numérica de buracos negros é dado por $n=\frac{ρ_{bn}}{m_{bn}}$

In [ ]:
m_bn= 1e-8
n= (rho_bn)/(m_bn)
print(n,'pc⁻³')

1005847.4788041 pc⁻³


Para unidade astronômica, transformamos essa densidade numérica fazendo: $\frac{1}{(206265)^{3} AU^{3}}$

In [ ]:
n1=(n/(206265**3))
print(n1,'AU⁻³')

1.1461870789033616e-10 AU⁻³


A distância mais próxima está relacionada com $n⋅V=1$ pois $n = \frac{N}{V}$ então $N=1$ $\therefore$ já que $ V(R)= \frac{4πR^{3}}{3}$  concluímos que a distância será $d= \sqrt[3]{\frac{3}{n4\pi}}$, daí:

In [ ]:
d = ((3)/(n1*4*pi))**(1/3)
print(d,'AU')

1277.0815405241792 AU


A frequência que esperamos encontrar um buraco negro dentro de 1AU está intimamente ligada a taxa de interação das partículas, que por sua vez depende fortemente da seção de choque. Sendo esta última $σ_{AU}= \pi \cdot d^{2}$ então $Γ=nσ_{AU}v$. Façamos isso para achar a frequência para uma distância de 1UA, daí:

In [ ]:
d1=1
sigma=pi*(d1**2)
sigma

3.14159265359

In [ ]:
gama=n1*sigma*v
print(gama,'yr⁻¹')

1.785006727255267e-08 yr⁻¹


Podemos ver essa frequência em Gyr de maneira que possamos comparar quantas vezes buracos negros já entraram neste raio desde o início do universo. Por isso fazemos $gama ⋅ \frac{10^{9} yr}{1 Gyr}$

In [ ]:
gama1= gama*1e9
print(gama1,'Gyr⁻¹')

17.850067272552668 Gyr⁻¹


Repitimos os mesmos passos agora supondo que a matéria escura seja feita de MACHOS de uma forma geral, mas percebemos que haverá mudanças a partir da densidade numérica e partimos daí.

Assumimos uma massa $10^{-3}M_{\odot}$. 

In [ ]:
m_machos= 1e-3
n2= (rho_bn)/(m_machos)
print(n2,'pc⁻³')

10.058474788041 pc⁻³


In [ ]:
n3=(n2/(206265**3))
print(n3,'AU⁻³')

1.1461870789033616e-15 AU⁻³


Então a distância do MACHOS mais próxima será:

In [ ]:
d_machos = ((3)/(n3*4*pi))**(1/3)
print(d_machos,'AU')

59276.87418110035 AU


Então a frequência que vemos MACHOS dentro de 1UA será:

In [ ]:
gama_machos=n3*sigma*v
print(gama_machos,'yr⁻¹')

1.7850067272552666e-13 yr⁻¹


In [ ]:
gama1_machos= gama_machos*1e9
print(gama1_machos,'Gyr⁻¹')

0.00017850067272552665 Gyr⁻¹


🐄 A galáxia Draco é uma galáxia anã dentro do Grupo Local. Sua luminosidade é $L = (1,8 ± 0,8)\times 10^{5} L_{⊙} $ e metade de sua luminosidade total é contido dentro de uma esfera de raio $r_{h} = 120 ± 12$ pc. A gigante vermelha
estrelas na galáxia Draco são brilhantes o suficiente para ter sua linha de visão
velocidades medidas. A dispersão de velocidade medida da gigante vermelha
estrelas na galáxia Draco é $σ_{r} = 10,5 ± 2,2 kms^{−1}$ .


   a) Qual a massa da galáxia Draco?
   
   b) Qual é a sua relação massa-luz? Descreva o
possíveis fontes de erro em sua estimativa de massa desta galáxia.

→ Partimos do teorema do Virial para um estado estacionário: $\frac{M<v^{2}>}{2}=\frac{αGM^{2}}{2r_{h}}$

onde $<v^{2}>$ é a dispersão de velocidade e ela representa $3σ^{2}$ pois consideramos uma esfera isotrópica. Assumimos que dentro do raio efetivo exista metade da massa (o que já causa uma problemática) e utilizamos α = 0.45 e ele depende do tipo de galáxia. 

Dito isso, isolamos a massa dentro do teorema do Virial e buscamos também determinar a incerteza ΔM envolvida no problema, $\therefore$ fazemos:

In [ ]:
sigma=10.5
rh=120
alpha=0.45
G=4.3e-3 #pc/Msol*(km/s)²

In [ ]:
M=(3*sigma**2*rh)/(alpha*G)
print(M,'Msol')

20511627.906976745 Msol


Para calcularmos a incerteza fazemos 

$(ΔM)^{2} =  (\frac{\partial M}{\partial \sigma_{r}} Δ\sigma_{r})^{2} + (\frac{\partial M}{\partial r_{h}} Δr_{h})^{2}$

sendo $\frac{\partial M}{\partial \sigma_{r}} = \frac{6r_{h}σ_{r}}{αG}$ e $\frac{\partial M}{\partial r_{h}}= \frac{3σ^{2}}{αG}$ e os acréscimos são $Δ\sigma_{r}= 2.2 kms^{-1}$ e $Δr_{h}=12 pc$ 

In [ ]:
d1=((6*rh*sigma*2.2)/(alpha*G))**2
d1


73880021633315.28

In [ ]:
d2=((36*sigma**2)/(alpha*G))**2
d2

4207268793942.671

In [ ]:
deltaM=np.sqrt((d1+d2))
print(deltaM,'Msol')

8836701.331789931 Msol


Portando, concluímos que $M \approx (2.05± 0.9)10^{7} M_{⊙}$

---



Razão massa/luminosidade?

🐄 Um raio de luz apenas roça a superfície da Terra $M = 6 × 10^{24} kg,
R = 6.4 × 10^{6} m$.

 Através de qual ângulo α o raio de luz é desviado pela lente gravitacional? (Ignore os efeitos de refração da atmosfera da Terra.) 

Repita seu cálculo para uma anã branca de $M = 2 × 10 ^{30} kg,
R = 1.5 × 10^{7}m$ e para uma estrela de nêutrons com $M=3×10^{30}kg, R+1.2×10^{4}m$.

→ Sabemos que o ângulo de deflexão seja $α=\frac{4GM}{c^{2}R}$, então para cada caso fazemos:

In [ ]:
G=6.67e-11
c=3e8

In [ ]:
#Para a Terra temos:

MT=6e24
RT=6.4e6

alphaT= (4*G*MT)/((c**2)*RT)
print(alphaT,'rad')

2.7791666666666667e-09 rad


In [ ]:
#Para anã branca

MWD=2e30
RWD=1.5e7

alphaWD= (4*G*MWD)/((c**2)*RWD)
print(alphaWD,'rad')


0.00039525925925925926 rad


In [ ]:
#Para estrela de neutrons 

MNS= 3e30
RNS=1.2e4

alphaNS=(4*G*MNS)/((c**2)*RNS)
print(alphaNS,'rad')

0.7411111111111112 rad


Vemos que se o objeto não for compacto, não possuirá grande ângulo de deflexão.

🐄 Vamos estimar a taxa de fusão de galáxias
no aglomerado Coma para ver se é realmente significativo. O Coma
contém N ≈ 1000 galáxias dentro de seu raio de meia massa de $r_{h} ≈
1.5 Mpc$. 

a) Qual é a densidade numérica média de galáxias dentro do
raio de meia massa? 

b) Suponha que a seção transversal típica de uma galáxia seja
$Σ ≈ 10^{−3} Mpc^{2}$ . Quão longe uma galáxia no aglomerado Coma viajará, em
média, antes de colidir com outra galáxia?

c) A dispersão da velocidade
do aglomerado Coma é $σ ≈ 880 kms^{ −3}$ . Qual é o tempo médio
entre colisões para uma galáxia no aglomerado Coma?

d) Este tempo é maior
maior ou menor que o tempo do Hubble?

→ Sabendo que as galáxias em aglomerados ricos são pobres em
velas padrão, porque elas tendem a ficar mais brilhantes com o tempo, uma vez que eventualmente elas se fundem com outras galáxia. 

Inicialmente a densidade numérica pode ser encontrada fazendo:

$n=\frac{N}{V}$ e conhecendo o volume sendo $ V = \frac{4πR^{3}}{3} $, daí:

In [ ]:
rh=1.5
pi=3.14159265359
N=1000

In [ ]:
V=(4*pi*(rh**3))/3
print(V,'Mpc³')

14.137166941155 Mpc³


In [ ]:
n= N/V
print(n,'Mpc⁻³')

70.73553026305994 Mpc⁻³


O livre caminho médio das galáxias no aglomerado COMA é a razão entre o volume e a seção transversal típica de uma galáxia $λ=\frac{V}{Σ}$.

Mas utilizamos a densidade numérica $n=\frac{N}{V}$ de uma galáxia para estimar o livre caminho médio, pois, para N=1 temos $ n =\frac{1}{Σλ}$ então finalmente $ λ =\frac{1}{Σn}$

In [ ]:
Sigma = 10**(-3) #Mpc²

Lambda = 1/(Sigma*n)
print(Lambda,'Mpc')

14.137166941155 Mpc


O tempo médio de colisão em uma certa direção envolve o livre caminho médio e a dispersão de velocidade das galáxias, de maneira que ele seja a razão entre essas duas variáveis. Então vemos $T=\frac{λ}{σ}$.

Mas, antes de aplicarmos os dados conhecidos faz-se necessário colocar essas variáveis na mesma unidade de medida para posteriormente deixar somente em termos de Gyr. Por isso fazemos: 

$T=\frac{λ}{σ}\frac{10^{6}pc}{1Mpc}\frac{3.086×10^{16}m}{1pc}\frac{1km}{10^{3}m}\frac{1yr}{3.156×10^{7}s}\frac{1Gyr}{10^{9}yr}$

In [ ]:
sigmav=880

pcm= 3.086*(np.power(10,16))
sup=np.power(10,6)*pcm
sup


3.086e+22

In [ ]:
yrs=3.156*(np.power(10,7))
inf=np.power(10,12)*yrs
inf

3.156e+19

In [ ]:
T= (Lambda*sup)/(sigmav*inf)
print(T,'Gyr')

15.708641973587227 Gyr


Encontramos, finalmente o tempo de colisão aproximado em 15.7 Gyrs, e concluímoms que em média não há colisão pois este tempo é maior do que o tempo de Hubble.